core/pbcc_memory_utils.c

```diff
@@ -0,0 +1,17 @@
+/* -*- c-file-style:"stroustrup"; indent-tabs-mode: nil -*- */
+#include "pbcc_memory_utils.h"
+
+#include <stdlib.h>
+
+
+// ----------------------------------------------
+//                   Functions
+// ----------------------------------------------
+
+void pbcc_free_ptr(void** ptr)
+{
+    if (NULL == ptr || NULL == *ptr) return;
+
+    free(*ptr);
+    *ptr = NULL;
+}
```

Comment:
This is quite risky approach. 

What in case whereas `ptr` points to the allocated memory. 
I know that propably not in your use-case but I'm aware of the future development and using this function on allocated pointer to allocated pointer (which is ofc very tricky and nobody should do this ofc) might fail. 

I can assume some array of pointers that might be failing here.

Reply:
I will remove this function because the task for which it has been done has been solved in another way :)